File: Ui/src/app/state/customer.model.ts
export interface Customer {
    customerId: number;
    firstName: string;
    lastName: string;
    emailAddress: string;
    updateDateTime: Date;
    createdDateTime: Date;
}

export interface CustomerState {
    selectedCustomer?: Customer;
    items: Customer[];
    pageSize: number;
    page: number;
    totalRows: number;
    totalPages: number;
    isLoading: boolean;
    isEndOfPage: boolean;
}

export interface CustomerSelectionState {
    customerId: number;
    isSelecting: boolean;
}

export interface CustomerPagination {
    pageSize: number;
    page: number;
    isLoading: boolean;
    isEndOfPage: boolean;
}

File: Api/Dynatron.Api/Models/CustomerModel.cs
using Dynatron.Domain;

namespace Dynatron.Api.Models
{
    public class CustomerModel
    {
        private CustomerModel() { }

        public int CustomerId { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string? EmailAddress { get; set; }
        public DateTimeOffset UpdateDateTime { get; set; }
        public DateTimeOffset CreatedDateTime { get; set; }

        public static CustomerModel FromEntity(Customer entity)
        {
            return new CustomerModel
            {
                CustomerId = entity.CustomerId,
                FirstName = entity.FirstName,
                LastName = entity.LastName,
                EmailAddress = entity.EmailAddress,
                CreatedDateTime = entity.CreatedDateTime,
                UpdateDateTime = entity.UpdatedDateTime
            };
        }
    }
}
